### PR TITLE
Multi-app

### PR DESCRIPTION
**Pull request by niallobyrnes**
I started with huntcsg suggestions for a multi-app "Site" and created this fork.

Theres an example in multi_app_example.py showing how it works.  With very little change to the original code, a Site is created, that creates a cherrypy tree of the apps.  A small method addApp is called on Site class to build the tree.  

The html renders, a number of tabs (actually they are probably buttons) across the top of the page, picking up an additional options class variable (app_bar_html) or uses the title or classname for the tab text.

The app nav bar styling needs some work.

### CODE REVIEW

**Comment by tramdas** [comment]
Thanks for this @niallobyrnes, was actually just thinking I needed this functionality soon.

**Comment by niallobyrnes** [comment]
Hi tramdas, it should be ready to merge and its got some examples included.  The css could do with some work as I just copied the tabs css (I dont know too much css).

**Comment by adamhajari** [comment]
@niallobyrnes this is awesome. I'll add some inline comments, but hope to merge this in asap.

File: examples/NBS_category_example.py [hunk 1]
```diff
@@ -5,93 +5,94 @@
 import matplotlib.pyplot as plt
 
 class NBSCategoriesApp(server.App):
-	title = "Spyre Example"
```
Comment (adamhajari) [inline examples/NBS_category_example.py]: looks like tabs got converted to spaces...

File: spyre/server.py [hunk 2]
```diff
@@ -27,296 +27,366 @@
...
+                self.templateVars['inputs'] = inputs
+            if controls is not None:
+                self.templateVars['controls'] = controls
+            if outputs is not None:
+                self.templateVars['outputs'] = outputs
+            if tabs is not None:
+                self.templateVars['tabs'] = tabs
+
+
+        self.getDataFrame = getDataFrameFunction
```
Comment (adamhajari) [inline spyre/server.py]: is getDataFrame used anywhere?

**Comment by adamhajari** [comment]
@niallobyrnes there were several merge conflicts with work I did recently to make spyre compatible with python3. I resolved those locally and have merged the bulk of your changes. Thanks!

**Pull request closed without merging**